src/components/actions/dev_tools/routes.ts

```diff
@@ -0,0 +1,264 @@
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
+
+    case 'ON_SCHEDULE':
+      if (!eventCode) return reply.send('eventCode required')
+      const { id, data } = await db.getScheduledEvent(applicationId, eventCode)
+      console.log(id)
+      triggerPayload.trigger = Trigger.OnSchedule
+      triggerPayload.table = 'trigger_schedule'
+      triggerPayload.record_id = id
+      triggerPayload.data = data
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
+
+    case 'ON_PREVIEW':
+      const revId = reviewId ?? (await getRandomReviewId(applicationId))
+      triggerPayload.trigger = Trigger.OnPreview
+      triggerPayload.table = 'review'
+      triggerPayload.record_id = revId
+      triggerPayload.applicationDataOverride = {
+        ...applicationDataOverride,
+        reviewData: {
+          latestDecision: {
+            decision: decision ?? Decision.Conform,
+            comment: comment ?? 'Test comment',
+          },
+        },
+      }
+
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId, reviewId: revId })
+      break
+
+    case 'ON_REVIEW_ASSIGN':
+      // If no assignment id, select one at random (using latest stage and level) and assign all possible sections to it
+      {
+        const assignment = assignmentId
+          ? await db.getSingleReviewAssignment(assignmentId)
+          : await selectRandomReviewAssignment(applicationId, sectionCodes)
+        if (!assignment.allowedSections) assignment.allowedSections = sectionCodes
+        if (sections)
+          assignment.allowedSections = assignment.allowedSections.filter((section: string) =>
+            sections.includes(section)
+          )
+        await db.assignReview(assignment)
+
+        triggerPayload.trigger = Trigger.OnReviewAssign
+        triggerPayload.table = 'review_assignment'
+        triggerPayload.record_id = assignment.id
+
+        actionsOutput = await processTrigger(triggerPayload)
+        finalApplicationData = await getApplicationData({
+          applicationId,
+          reviewAssignmentId: assignmentId,
+        })
+      }
+      break
+
+    case 'ON_REVIEW_CREATE':
+      // Get assignment id -- either passed in, or pick one at random (assigned and highest stage/level)
+      {
+        const assignment = assignmentId
+          ? await db.getSingleReviewAssignment(assignmentId)
+          : await selectRandomReviewAssignment(applicationId, sectionCodes, true)
+        // Create a review record and review_decision record
+        const { id: revId } = await db.createReview(assignment.id)
+        await db.createReviewDecision(revId)
+        // Create one review response
+
+        // TO-DO: Make some review responses and give them dummy values?
```

Comment:
Oh here is the response to my question earlier on

Reply:
Yeah, I wasn't sure if it would be needed. I don't think it is. I haven't yet come across a scenario for trigger testing where the presence of real review_responses turned out to actually make a difference. Please let me know if you find one.

Reply:
I was thinking about how to check LOQ selected for different review levels. Then it would require to check the review_response and their decision, but I think that's a very specific event check

Reply:
I'll be interested to see if the configs that check review response counts cause problems with this. If they do, I can revisit it.